### ISSUE TITLE
Old left rooms reappear in broken state

### ISSUE THREAD

**Issue by emgrav** [famedly/matrix-dart-sdk#1916]
### Checklist

- [X] I could not find a solution in the documentation, the existing issues or discussions.
- [X] I already asked for help in the chat

### In which Project did the bug appear?

Famedly App

### On which platform did the bug appear?

Firefox

### SDK Version

Unknown, but app v2.39.1-0-g0b66f4b1 

### Describe the problem caused by this bug

I logged into my account using element-desktop. After this, a dozen old left rooms from the archive appeared at the top of my room list in Famedly. The rooms are not visible in Element, and appear to be in a broken state. The "last activity" indicator in the room list is always the current time, meaning the rooms are pinned to the top of the room list sorted by activity. The rooms are all called "Empty chat", and attempting to leave them does nothing. Attempting to send a message in the rooms set the last activity to an actual timestamp, unpinning them from the room list. Sending the even did however not succeed.

Nico theorized that element did something with the room archive and account data which caused this to happen.

### Steps To Reproduce

1. Have old rooms from 2022 and 2023 in the archive
2. Log into element-desktop

**Comment by nico-famedly** [comment]
I have also experienced that sometimes when opening the archive, possibly some account data events cause the room to be populated? Needs some debugging.

**Comment by Clemens-Toegel** [comment]
We also experience this problem with our client. In our setup no other client is involved which may cause a problem, we only tested with the matrix-dart-sdk.
The behaviour is the following: In the same session no bug occurs, we can leave a room (lets say _Room A_) and find it then in the room list as archived room with membership leave. After a new login, however, we see  _Room A_ as 'Empty chat' room in our rooms list instead of in the archived rooms list. After debugging I have noticed that the left room _Room A_ is in the archived rooms list as expected, but also in the rooms list (with membership joined!), which is why we see _Room A_ in the rooms list with title 'Empty chat' and membership joined.

I have pinned this behaviour down to the version V30.0, before this version we don't see such a behaviour.

Having a look at the changelog I may think that https://github.com/famedly/matrix-dart-sdk/pull/1866 introduced this behaviour. Specifically, this change:
https://github.com/famedly/matrix-dart-sdk/blob/4fd616bcbf483104d6a1be3c5b3b295b7687b87c/lib/encryption/key_manager.dart#L204

The `JoinedRoomUpdate` may cause the room to be added in the rooms list where it does not belong.
